pytest_inmanta/plugin.py

```diff
@@ -289,17 +314,14 @@ def project_factory(
             raise
 
     with open(os.path.join(project_dir, "project.yml"), "w+") as fd:
-        metadata: typing.Mapping[str, object] = {
-            "name": "testcase",
-            "description": "Project for testcase",
-            "repo": repos,
-            "modulepath": modulepath,
-            "downloadpath": "libs",
-            "install_mode": install_mode.value,
-        }
-        yaml.dump(metadata, fd)
+        # pydantic.BaseModel.dict() doesn't produce data that can be serialized
+        # so we first serialize it as json, then load it and dump it as yaml.
+        yaml.safe_dump(json.loads(project_metadata.json()), fd)
 
-    ensure_current_module_install(os.path.join(project_dir, "libs"), in_place)
+    ensure_current_module_install(
+        os.path.join(project_dir, "libs"),
+        in_place=inm_mod_in_place.resolve(request.config),
```

Comment:
Why not use the variable anymore?

Reply:
Because it is not used in another place anymore (the other logic has been moved to the project_metadata fixture)